src/Groups/schur_index.jl

```diff
@@ -0,0 +1,520 @@
+
+  # check for the structure U:P where U is a normal subgroup of odd prime
+  # order r and P is a 2-group
+  facts = sort(collect(factor(n)))
+  length(facts) == 2 || return no_type
+  (facts[1][1] == 2 && facts[2][2] == 1) || return no_type
+  r = facts[2][1]
+  U = sylow_subgroup(G, r)[1]
+  is_normalized_by(U, G) || return no_type
+
+  # check for type ("Q8", r), i.e.,
+  # X = C_P(U) is isomorphic with Q8,
+  # P/X is nontrivial and cyclic,
+  # P is a central product of X and C_P(X),
+  # and |P/X| is the 2-part of the mult. order of 2 modulo r.
+  P = sylow_subgroup(G, 2)[1]
+  X = centralizer(P, U)[1]
+  x = order(X)
+  mord = modord(ZZRingElem(2), r)
+  twopart, mord = ppio(mord, 2)
+  p = order(P)
+  if x == 8 && small_group_identification(X) == id_q8 &&
+     p != x && x * twopart == p
+    Q = quo(P, X)[1]
+    if is_cyclic(Q)
+      cpx = centralizer(P, X)[1]
+      inter = intersect(X, cpx)
+      C = sub(P, vcat(gens(X), gens(cpx)))[1]
+      is_subset(inter[1], center(P)[1]) && order(C) == p && return ("Q8", r)
+    end
+  end
+
+  # check for type ("QD", r), i.e.,
+  # Z = P' is cyclic of order at least 4,
+  # Y = C_P(z) satisfies Y/Z cyclic, where z is an element of order 4 in Z,
+  # X = C_P(U) is not abelian,
+  # Z has index 2 in X,
+  # and the order of 2 modulo r has the form 2^s f where f is a (not nec. odd)
+  # integer and |P/X| = 2^s.
+  # (We exclude groups where f is odd and X is isomorphic with Q8;
+  # these have been detected above with result ("Q8", r),
+  # hence we need not check this property.)
+  Z = derived_subgroup(P)[1]
+  (is_cyclic(Z) && order(Z) >= 4) || return no_type
+  z = cyclic_generator(Z)^div(order(Z), 4)
+  Y = centralizer(P, z)[1]
+  is_cyclic(quo(Y, Z)[1]) || return no_type
+  is_abelian(X) && return no_type
+  (order(X) == 2*order(Z)) || return no_type
+  (mod(twopart, div(order(P), order(X))) == 0) || return no_type
+  return ("QD", r)
+end
+
+
+#############################################################################
+#
+# deal with extensions of p-adic fields by elements of abelian number fields
+#
+# For `vals` with conductor N = p^k m, with m coprime to p,
+# we describe Q_p(vals) by (N, stab, [Q_p(N):Q_p]),
+# where stab is the set of prime residues u modulo N
+# such that *u is in Gal(Q_p(N)/Q_p) and chi is fixed by *u
+#
+# We use that Gal(Q_p(N)/Q_p) is the direct product of the group generated
+# by \sigma_p: \zeta_m \mapsto \zeta_m^p, \zeta_{p^k} \mapsto \zeta_{p^k}
+# and the group of maps
+# \sigma_u: \zeta_m \mapsto \zeta_m, \zeta_{p^k} \mapsto \zeta_{p^k}^u,
+# where u is a prime residue modulo p^k.
+#
+# This means that [Q_p(vals):Q_p] == [Q_p(N):Q_p] / length(stab)
+#
+const _Q_p_chi_cache = Dict{Tuple{Int, Vector{Int}, Int},
+                            Tuple{Int, Vector{Int}, Int}}()
+
+function _Q_p_chi(vals::GapObj, p::ZZRingElem)
+  if length(vals) == 0
+    F = GAP.Globals.Rationals
+  else
+    F = GAP.Globals.Field(GAP.Globals.Rationals, vals)
```

Comment:
we need to talk. givem the amout of GapField stuff in here, we seem to be missing canonical useful stuff in Oscar direct....I really glad it works, but here there seems to be work required (pos. by me)